instrumentation/apache-camel-2.20/src/test/groovy/test/TwoServicesCamelSpringBootBasedTest.groovy

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package test
+
+import static io.opentelemetry.trace.Span.Kind.CLIENT
+import static io.opentelemetry.trace.Span.Kind.INTERNAL
+import static io.opentelemetry.trace.Span.Kind.SERVER
+
+import com.google.common.collect.ImmutableMap
+import io.opentelemetry.auto.test.AgentTestRunner
+import io.opentelemetry.auto.test.utils.PortUtils
+import io.opentelemetry.trace.attributes.SemanticAttributes
+import org.apache.camel.CamelContext
+import org.apache.camel.ProducerTemplate
+import org.apache.camel.builder.RouteBuilder
+import org.apache.camel.impl.DefaultCamelContext
+import org.springframework.boot.SpringApplication
+import org.springframework.context.ConfigurableApplicationContext
+import spock.lang.Shared
+
+class TwoServicesCamelSpringBootBasedTest extends AgentTestRunner {
+
+  @Shared
+  int portOne = PortUtils.randomOpenPort()
+  @Shared
+  int portTwo = PortUtils.randomOpenPort()
+  @Shared
+  ConfigurableApplicationContext server
+  @Shared
+  CamelContext clientContext
+
+  def setupSpec() {
+    createServer()
+  }
+
+  def createServer() {
+    def app = new SpringApplication(TwoServicesConfig)
+    app.setDefaultProperties(ImmutableMap.of("service.one.port", portOne, "service.two.port", portTwo))
+    server = app.run()
+  }
+
+  def createAndStartClient() {
+    clientContext = new DefaultCamelContext()
+    clientContext.addRoutes(new RouteBuilder() {
+      void configure() {
+        from("direct:input")
+          .log("SENT Client request")
+          .to("http://localhost:$portOne/serviceOne")
+          .log("RECEIVED Client response")
+      }
+    })
+    clientContext.start()
+  }
+
+  def cleanupSpec() {
+    if (server != null) {
+      server.close()
+      server = null
+    }
+  }
+
+  def "two camel service spans"() {
+    setup:
+    createAndStartClient()
+    ProducerTemplate template = clientContext.createProducerTemplate()
+
+    when:
+    template.sendBody("direct:input", "Example request")
+
+    then:
+    assertTraces(1) {
+      trace(0, 9) {
+        it.span(0) {
+          name "input"
+          kind CLIENT
+          attributes {
+            "camel.uri" "direct://input"
+          }
+        }
+        it.span(1) {
+          name "input"
+          kind INTERNAL
+          attributes {
+            "camel.uri" "direct://input"
+          }
+        }
+        it.span(2) {
+          name "POST"
+          kind CLIENT
+          attributes {
+            "$SemanticAttributes.HTTP_METHOD.key" "POST"
+            "$SemanticAttributes.HTTP_URL.key" "http://localhost:$portOne/serviceOne"
+            "$SemanticAttributes.HTTP_STATUS_CODE.key" 200
+            "camel.uri" "http://localhost:$portOne/serviceOne"
+          }
+        }
+        it.span(3) {
+          name "HTTP POST"
+          kind CLIENT
```

Comment:
Having two `CLIENT` spans in a row looks wrong -- [semantic convention](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#spankind) says that `CLIENT` must be a parent of `SERVER` span. 
Can you merge those two spans into one?

Reply:
It's not something that Camel instrumentation can control. The second CLIENT span is generated by "commons-http-client" instrumentation that kicks in the middle of execution (ie is called by camel implementation).

Reply:
Then camel instrumentation should not generate the first span in the first place, I believe

Reply:
Then you're loosing data that can be valuable for customers. Also, Camel instr does not know if any internals used to execute the call are instrumented and will generate the span or not. How would you approach that?

Reply:
As discussed offline, two ideas:
- create Camel spans as INTERNAL, counting on another instrumentation creating CLIENT just before sending a message 
- change some of our "last stage before sending" instrumentations (in this case - http client) to check if parent span was CLIENT - if so, change it's type to become INTERNAL

@trask @iNikem any thoughts on the above?

Reply:
I believe there is a separate issue to handle such cases (https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/465). Will leave this as it is then.